modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -630,201 +646,222 @@ private CompletableFuture<?> createTablePartitionsLocally(
 
             CompletableFuture<?>[] futures = new CompletableFuture<?>[partitions];
 
-            // TODO: https://issues.apache.org/jira/browse/IGNITE-19713 Process assignments and set partitions only for assigned partitions.
-            PartitionSet parts = new BitSetPartitionSet();
+            PartitionSet partitionSet = new BitSetPartitionSet();
 
+            // TODO: https://issues.apache.org/jira/browse/IGNITE-19713 Process assignments and set partitions only for assigned partitions.
             for (int i = 0; i < futures.length; i++) {
                 futures[i] = new CompletableFuture<>();
 
-                parts.set(i);
+                partitionSet.set(i);
             }
 
-            String localMemberName = localNode().name();
-
             for (int i = 0; i < partitions; i++) {
                 int partId = i;
 
                 Set<Assignment> newPartAssignment = newAssignments.get(partId);
 
-                InternalTable internalTbl = table.internalTable();
+                // TODO https://issues.apache.org/jira/browse/IGNITE-19170 #handleChangePendingAssignmentEvent should be called on
+                // TODO actual event, the method #createTablePartitionsLocally should be removed.
+                handleChangePendingAssignmentEvent(
```

Comment:
What's the point in considering initial assignments as pending one? Generally it sounds reasonable but only if you took into consideration the fact that assignments.stable might be empty and in this case there's no sense in calling changePeersAsync, did you?